File: zhanglianzhuang/day2/demo02/src/main/java/com/example/demo02/service/imp/RoleServiceImp.java
package com.example.demo02.service.imp;

import com.example.demo02.dao.IRole;
import com.example.demo02.entity.Role;
import com.example.demo02.service.RoleService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class RoleServiceImp implements RoleService {

	/*自动装载*/
	@Autowired
	private IRole iRole;
	@Override
	public int insertRole(Role role) {
		return iRole.insertRole(role);
	}

	@Override
	public List<Role> getRole(Role role) {
		return iRole.getRole(role);
	}

	@Override
	public int updateRole(Role role) {
		return iRole.updateRole(role);
	}

	@Override
	public int deleteRole(Role role) {
		return iRole.deleteRole(role);
	}

	@Override
	public int getRolebyname(Role role){
		return iRole.getRolebyname(role);
	}
}

File: caiguicheng/day03/work/demo03work/service/imp/RoleServiceImp.kt
package com.example.demo03work.service.imp

import com.example.demo03work.dao.IRole
import com.example.demo03work.entity.Role
import com.example.demo03work.entity.User
import com.example.demo03work.service.RoleService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Service

@Service
class RoleServiceImp:RoleService {
    //使用时候创建
    @Autowired
    @Lazy
    lateinit var irole: IRole

    /**
     * 插入角色
     * @return 是否成功
     */
    override fun insertRole(role: Role): Int{
        return irole.insertRole(role)
    }

    /**
     * 更新角色信息
     * @return 是否成功
     */
    override fun updateRole(role: Role): Int{
        return irole.updateRole(role)
    }

    /**
     * 根据条件获得角色
     * @return 用户列表
     */
    override fun selectRole(role: Role):List<Role>{
        return irole.selectRole(role)
    }

    /**
     * 删除角色
     * @return 是否成功
     */
    override fun deleteRole(id:Int):Int{
        return irole.deleteRole(id)
    }
}